src/util/file.c

```diff
@@ -358,6 +358,30 @@ int fileput(char *path, char *string) {
     return(0);
 }
 
+int fileputat(int fd, char *path, char *string) {
```

Comment:
@vsoch @gmkurtzer I offered to tackle this issue, however I'm just back at uni so I wasn't working the last few weeks. I will handle the continuation of shifting to file descriptors, however I think we can merge the new environment format before that is all finished.